### ISSUE TITLE
Automatizar geração de documentação

### ISSUE THREAD

**Issue by diogommartins** [async-worker/async-worker#20]
Tava buscando por formar de automatizar esse fluxo e encontrei essa actionÇ https://github.com/marketplace/actions/sphinx-build 

> If you have any Python dependencies that your project needs (themes, build tools, etc) then place them in a requirements.txt file inside your docs folder.

👎 Achei isso de ter que replicar a listagem de dependências meio chato, mas aceitável, já que isso tende a mudar muito pouco.
👍 Erros de documentação direto no PR

**Comment by daltonmatos** [comment]
Eu sempre pensei em poder ter esses checks na doc também. Queria inclusive considerar todos os warnings como erros. Mas nunca consegui me entender com os warnings que o sphinx sempre mostra quando fazemos build. 
Penso que o primeiro passo seria conseguir rodar o build das docs passando as opções de "considera warning como erro" e depois podemos começar a ter esse workflow, o que acha?

Um ponto interessante é fazer um filtro nesse workflow (quando ele passar a existir) para que ele rode apenas quando houver mudanças nos arquivos dentro de `docs-src/`.

**Comment by diogommartins** [comment]
> Eu sempre pensei em poder ter esses checks na doc também. Queria inclusive considerar todos os warnings como erros. Mas nunca consegui me entender com os warnings que o sphinx sempre mostra quando fazemos build.
> Penso que o primeiro passo seria conseguir rodar o build das docs passando as opções de "considera warning como erro" e depois podemos começar a ter esse workflow, o que acha?

Concordo, acho que é um bom primeiro passo. Vou dar uma lida no código fonte dessa action pra entender como eles fizeram isso.


> Um ponto interessante é fazer um filtro nesse workflow (quando ele passar a existir) para que ele rode apenas quando houver mudanças nos arquivos dentro de `docs-src/`.

Só o docs-src é suficiente? As docsstrings de classes e métodos não vão parar em nenhum lugar na documentação ?

**Comment by daltonmatos** [comment]
Se não me engano tem um argumento do sphinx pra considerar tudo como erro.

Em relação às docstrings, acho que a versão html vai pro `docs/` sim.

A ideia é que os checks de documentação façam alguma conferência nas docstrings também?

Achei que a ideia era só pra confirmar que "seu PR consegue fazer um build clean de toda a doc".
